vlib/v/gen/c/testdata/c_fn_args.vv

```diff
@@ -0,0 +1,9 @@
+fn foo(bar &char, baz &char) {
+	println(unsafe { bar.vstring() + baz.vstring() })
+}
+
+fn main() {
+	baz := 'baz'.str
+	// Should compile with `-cc [gcc/clang] -cstrict`
+	foo('bar'.str, baz)
+}
```

Comment:
No, it should not compile with -cstrict. Such hacks are brittle. 

It should need explicit casts, like this: `foo(&char('bar'.str), &char(baz))`.

Reply:
but why should the user do the explicit cast with flags? I feel they don't need to do it.

Reply:
I am not sure what flags are you talking about?

Reply:
`-cstrict`

Reply:
The user should have done the explicit cast in *all cases*, `-cstrict` just makes the C compilers that support it much more strict, and makes them produce more errors for situations, that normally they accept.